libs/clients/syslumenn/src/lib/syslumennClient.config.ts

```diff
@@ -0,0 +1,26 @@
+import { defineConfig } from '@island.is/nest/config'
+import * as z from 'zod'
+
+const schema = z.object({
+  url: z.string(),
+  fetch: z.object({
+    timeout: z.number().int(),
+  }),
+  username: z.string(),
+  password: z.string(),
+})
+
+export const SyslumennClientConfig = defineConfig({
+  name: 'SyslumennApi',
+  schema,
+  load(env) {
+    return {
+      url: env.required('SYSLUMENN_HOST', 'https://api.syslumenn.is/Test'),
+      fetch: {
+        timeout: env.optionalJSON('SYSLUMENN_TIMEOUT') ?? 10000,
+      },
+      username: env.required('SYSLUMENN_USERNAME', 'SYSLUMENN_USERNAME'),
+      password: env.required('SYSLUMENN_PASSWORD', 'SYSLUMENN_PASSWORD'),
```

Comment:
Should not have "fake" devFallback here. Should just skip the second argument if there's nothing we want to commit to the repo.

Our ConfigModule is designed to throw a "friendly" error when someone accesses a config which is missing some values. The annoying thing is if someone tries to read the config while NestJS is starting up, it will crash the process until you add the missing environment variables.

For ideal DX, we only read the config when needed. In many cases this can be tricky, but in this case it might be simple:

If we move SyslumennService to the client module, then I think we should completely wrap the Api class and construct it inside SyslumennService's login function. Only export SyslumennService. Then we don't even need AuthMiddleware or HeadersMiddleware, as you can pass accessToken to the Api constructor.